testing/integration/k8s/agent_helm_test.go

```diff
@@ -0,0 +1,212 @@
+			"kubernetes.containers.logs.enabled=true",
+
+			// Disable others
+			"kubernetes.state.enabled=false",
+			"kubernetes.metrics.enabled=false",
+			"kubernetes.apiserver.enabled=false",
+			"kubernetes.proxy.enabled=false",
+			"kubernetes.scheduler.enabled=false",
+			"kubernetes.controller_manager.enabled=false",
+			"kubernetes.containers.metrics.enabled=false",
+			"kubernetes.containers.state.enabled=false",
+			"kubernetes.containers.audit_logs.enabled=false",
+			"kubernetes.pods.enabled=false",
+		},
+	}
+
+	steps := []k8sTestStep{
+		k8sStepCreateNamespace(),
+
+		// 1st - deploy flog
+		func(t *testing.T, ctx context.Context, kCtx k8sContext, namespace string) {
+			k8sStepDeployApp("flog.yaml")(t, ctx, kCtx, namespace)
+		},
+
+		// 2nd - deploy the agent without rotated logs enabled
+		k8sStepHelmDeployWithValueOptions(AgentHelmChartPath, "elastic-agent",
+			defaultValues,
+		),
+
+		// 3rd - check that the agent pod is running
+		k8sStepCheckRunningPods(
+			"name=agent-pernode-elastic-agent",
+			1, "agent"),
+
+		// 4th - verify logs are ingested from `/var/log/containers/`
+		k8sStepCheckLogFilesIngested(info,
+			"logs", "kubernetes.container_logs", "default", "/var/log/containers/*flog*.log",
+			expectedLogFile{regex: containerRegex,
+				description: "container log (" + containerRegex.String() + ")"},
+		),
+
+		// 5th - upgrade the agent to enable rotated logs
+		k8sStepHelmUpgrade(AgentHelmChartPath, "elastic-agent",
+			values.Options{
+				ValueFiles: defaultValues.ValueFiles,
+				Values: append(defaultValues.Values,
+					"kubernetes.containers.logs.rotated_logs=true"),
+			}),
+
+		// 6th - check that the agent pod is running
+		k8sStepCheckRunningPods(
+			"name=agent-pernode-elastic-agent",
+			1, "agent"),
+
+		// 7th - verify rotated logs are ingested
+		k8sStepCheckLogFilesIngested(info,
+			"logs", "kubernetes.container_logs", "default", "/var/log/pods/*flog*",
+			expectedLogFile{regex: plainRegex,
+				description: "plain text rotated log (" + plainRegex.String() + ")"},
+			expectedLogFile{regex: gzRegex,
+				description: "gzipped rotated log (" + gzRegex.String() + ")"},
+		),
+	}
+
+	ctx := context.Background()
+	testNamespace := kCtx.getNamespace(t)
+
+	for _, step := range steps {
+		step(t, ctx, kCtx, testNamespace)
+	}
+}
+
+// expectedLogFile represents a log file pattern to verify in Elasticsearch
+type expectedLogFile struct {
+	regex       *regexp.Regexp
+	description string
+}
+
+// k8sStepCheckLogFilesIngested creates a test step that verifies rotated logs are ingested
+// by querying Elasticsearch and checking for files matching the provided regex patterns.
+func k8sStepCheckLogFilesIngested(
+	info *define.Info,
+	dsType, dataset, datastreamNamespace, wildcardPath string,
+	expectedFiles ...expectedLogFile,
+) k8sTestStep {
+	return func(t *testing.T, ctx context.Context, kCtx k8sContext, namespace string) {
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
```

Comment:
```suggestion
		require.EventuallyWithT(t, func(collectT *assert.CollectT) {
```

This is dangerous shadowing imo, even if it works. If you do rename it, be careful to rename all the uses inside the test function.